test/functional-api/integration/aspect/aspect-templates.it-spec.ts

```diff
@@ -0,0 +1,600 @@
+      expect(data).toEqual(aspectDataCreate);
+    });
+  });
+
+  describe('Update Aspect template already utilized by an aspect', () => {
+    let aspectTypeFromHubTemplate = '';
+    beforeAll(async () => {
+      const resAspectonHub = await createAspectNewType(
+        entitiesId.hubContextId,
+        templateType,
+        `new-aspect-d-name-${uniqueId}`,
+        `new-aspect-name-id-${uniqueId}`
+      );
+      aspectDataCreate = resAspectonHub.body.data.createAspectOnContext;
+      hubAspectId = resAspectonHub.body.data.createAspectOnContext.id;
+      aspectTypeFromHubTemplate =
+        resAspectonHub.body.data.createAspectOnContext.type;
+    });
+    afterAll(async () => {
+      await removeAspect(hubAspectId);
+    });
+    test('Create aspect with existing aspect template, and update template type, doesnt change the aspect type', async () => {
+      // Act
+      await updateAspectTemplate(aspectTemplateId, templateType + ' - Update');
+
+      const getAspectsData = await aspectDataPerContext(
+        0,
+        entitiesId.hubId,
+        entitiesId.challengeId,
+        entitiesId.opportunityId
+      );
+      const data = getAspectsData[0];
+
+      // Assert
+      expect(aspectTypeFromHubTemplate).toEqual(templateType);
+      expect(data).toEqual(aspectDataCreate);
+    });
+
+    test('Update aspect to use the new aspect template type', async () => {
+      // Act
+
+      const resAspectonHub = await updateAspect(
+        hubAspectId,
+        aspectNameID,
+        aspectDisplayName + 'EA update',
+        aspectDescription + 'EA update',
+        templateType + ' - Update'
+      );
+      const aspectDataUpdate = resAspectonHub.body.data.updateAspect;
+      const aspectTypeFromHubTemplate =
+        resAspectonHub.body.data.updateAspect.type;
+      hubAspectId = resAspectonHub.body.data.updateAspect.id;
+
+      const getAspectsData = await aspectDataPerContext(
+        0,
+        entitiesId.hubId,
+        entitiesId.challengeId,
+        entitiesId.opportunityId
+      );
+      const data = getAspectsData[0];
+
+      // Assert
+      expect(aspectTypeFromHubTemplate).toEqual(templateType + ' - Update');
+      expect(data).toEqual(aspectDataUpdate);
+    });
+  });
+
+  describe('Remove Aspect template already utilized by an aspect', () => {
+    let aspectTypeFromHubTemplate = '';
+    beforeAll(async () => {
+      const resAspectonHub = await createAspectNewType(
+        entitiesId.hubContextId,
+        templateType,
+        `rem-temp-aspect-d-name-${uniqueId}`,
+        `rem-temp-aspect-name-id-${uniqueId}`
+      );
+      aspectDataCreate = resAspectonHub.body.data.createAspectOnContext;
+      hubAspectId = resAspectonHub.body.data.createAspectOnContext.id;
+      aspectTypeFromHubTemplate =
+        resAspectonHub.body.data.createAspectOnContext.type;
+    });
+    afterAll(async () => {
+      await removeAspect(hubAspectId);
+    });
```

Comment:
This is the same data required for the previous describe block, can we make a surrounding  describe with this data setup?